includes/pages/guest_login.php

```diff
@@ -119,6 +121,10 @@ function guest_register()
             $msg .= error(__('Please enter a nickname.'), true);
         }
 
+        if ($request->has('mobile_show') && $enable_mobile_show) {
```

Comment:
This implies that the consent will be revoked on save if the setting gets disabled temporarily and users update their consent setting